### ISSUE TITLE
Document API returns internal error 500 for empty body

### ISSUE THREAD

**Issue by aburmeis** [arangodb/arangodb#16604]
## My Environment

* __ArangoDB Version__:   3.9.1 Community Edition
* __Deployment Mode__:   Cluster
* __Deployment Strategy__:   Kubernetes

## Component, Query & Data

__Affected feature__:

Documents API

__Replication Factor & Number of Shards (Cluster only)__:
Replication Factor 3, force one shard

## Steps to reproduce

The following API call (regardless of database and document type) fails with an internal error:

```
curl 'http://arango-server-ea:8529/_db/TestDB/_api/document/TestEntities?onlyget=true' -X PUT --data '[]'
```

(add anycredentials using -u  if needed)

__Problem__:

The response is 
```
{"code":500,"error":true,"errorMessage":"internal error","errorNum":4}
```

__Expected result__:

I would expect a response with no documents.

**Comment by aburmeis** [comment]
This kind of call is also created using Java Driver 6.16 `ArangoCollection.getDocuments()` and Arango Spring Data 3.7.0 for `CrudRepository.findAllById()`.

**Comment by aburmeis** [comment]
It seems the error is **only** occurring in a **cluster** setup, the call is working for a single server.

**Comment by Simran-B** [comment]
Yes, there is a discrepancy between single server and cluster, and it has been addressed in v3.11.1, where the Document API returns an empty array instead of an error in the cluster case:

https://www.arangodb.com/docs/stable/release-notes-api-changes311.html#:~:text=%22_gG9JHsW%2D%2D%2D%22%7D-,Introduced%20in%3A%20v3.11.1,Using%20the%20Document%20API,-for%20reading%20multiple
